libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo/ast/internal/semanticEquals.kt

```diff
@@ -11,18 +11,25 @@ internal fun GQLNode.semanticEquals(other: GQLNode?): Boolean {
   return toSemanticSdl() == other.toSemanticSdl()
 }
 
+/**
+ * Normalizes the given [GQLNode].
+ *
+ * Two nodes with the same normalized representation behave the same during execution.
+ *
+ * Especially, descriptions are removed and field, inputFields, arguments, etc.. are sorted lexicographically.
+ */
 internal fun GQLNode.toSemanticSdl(): String {
   return transform2 { node ->
     when (node) {
       is GQLInputValueDefinition -> node.copy(description = null)
-      is GQLObjectTypeDefinition -> node.copy(description = null)
+      is GQLObjectTypeDefinition -> node.copy(description = null, fields = node.fields.sortedBy { it.name })
```

Comment:
Should we also sort the `implementsInterfaces` and `directives` to be safe? (Does the order of directives count? 🤔)

Reply:
Yes, we should 👍 , adding this now